tests/integration/acp.go

```diff
@@ -0,0 +1,66 @@
+// Copyright 2024 Democratized Data Foundation
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.txt.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0, included in the file
+// licenses/APL.txt.
+
+package tests
+
+import (
+	"github.com/sourcenetwork/immutable"
+	"github.com/stretchr/testify/require"
+)
+
+// AddPolicy will attempt to add the given policy using defraDB's ACP module.
+type AddPolicy struct {
+	// NodeID may hold the ID (index) of the node we want to add policy to.
+	//
+	// If a value is not provided the policy will be added in all nodes.
+	NodeID immutable.Option[int]
+
+	// The raw policy string.
+	Policy string
+
+	// The policy creator, i.e. actor creating the policy.
+	Creator string
```

Comment:
suggestion:  This would be much easier to maintain if instead of the string value an index (immutable.Option[int]) was provided, especially for the other actions (such as CreateDoc in the PR description).

e.g.
```
AddPolicy{
	Creator: 1,
},
CreateDoc{
	Identity: 1,
	Doc: `{"name": "John", "age": 27 }`,
},
```
The string value does not appear to be important to these tests, and it saves us adding another `cid`/`schemaVersionID` problem to the pile.

Reply:
That exactly what I have in the implementation. The Identity part is not ready for review. I will push those changes as they make it to tests.

Reply:
> due to time constraint for reworking all the tests

I do not think we should rush the review due to pressure from the business side of things if that is what you meant.

And this is a maintenance and [review] readability issue introduced in this PR.

Interested as to what others' opinions are here.

Reply:
This is nice suggestion but I don't see this as a blocker that would effect the readability in a massive way. One might argue that seeing the concrete signatures is more readable as we have often made the tests be as verbose as possible with limited fixtures. This does cause extra time when reworking the full authentication of Identities that would need to adjust with it likely. I don't see this as an important one to do so for PR, and non-blocker (as is a suggestion so I don't think you do either?)